### PR TITLE
SDL2_image: 2.0.5 -> 2.6.3

### PR DESCRIPTION
**Pull request by superherointj**
* SDL2_image: 2.0.5 -> 2.6.3
* python310Packages.pygame: pin SDL2_image to 2.0.5
  - Upstream decision: https://github.com/pygame/pygame/issues/3430#issuecomment-1279955564

Fixes #187685

Release: https://github.com/libsdl-org/SDL_image/releases/tag/release-2.6.3

This PR is ready to be reviewed/merged.

### CODE REVIEW

**Comment by wegank** [comment]
@ofborg build python310Packages.pygame

**Comment by superherointj** [comment]
@ofborg build python310Packages.pygame

**Comment by superherointj** [comment]
Result of `nixpkgs-review pr 215200` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>aqemu</li>
    <li>python310Packages.rl-coach</li>
  </ul>
</details>
<details>
  <summary>1 package blacklisted:</summary>
  <ul>
    <li>appimage-run-tests</li>
  </ul>
</details>
<details>
  <summary>15 packages failed to build:</summary>
  <ul>
    <li>cot (python310Packages.cot)</li>
    <li>d1x-rebirth-full</li>
    <li>d2x-rebirth-full</li>
    <li>koboredux</li>
    <li>ocamlPackages.ocaml-freestanding</li>
    <li>pureref</li>
    <li>python311Packages.cot</li>
    <li>python311Packages.pyautogui</li>
    <li>python311Packages.pygame</li>
    <li>python311Packages.pygame-gui</li>
    <li>python311Packages.pygetwindow</li>
    <li>python311Packages.pyrect</li>
    <li>python311Packages.pytmx</li>
    <li>solo5</li>
    <li>toppler</li>
  </ul>
</details>
<details>
  <summary>238 packages built:</summary>
  <ul>
    <li>SDL2_image</li>
    <li>Sylk</li>
    <li>_20kly</li>
    <li>abbaye-des-morts</li>
    <li>alephone</li>
    <li>alephone-durandal</li>
    <li>alephone-eternal</li>
    <li>alephone-evil</li>
    <li>alephone-infinity</li>
    <li>alephone-marathon</li>
    <li>alephone-pathways-into-darkness</li>
    <li>alephone-pheonix</li>
    <li>alephone-red</li>
    <li>alephone-rubicon-x</li>
    <li>alpine-make-vm-image</li>
    <li>altair</li>
    <li>anbox</li>
    <li>anki-bin</li>
    <li>anytype</li>
    <li>appimage-run</li>
    <li>assaultcube</li>
    <li>beekeeper-studio</li>
    <li>betterdiscord-installer</li>
    <li>blobby</li>
    <li>blobwars</li>
    <li>blockattack</li>
    <li>bloomrpc</li>
    <li>bootstrap-studio</li>
    <li>brutalmaze</li>
    <li>buttercup-desktop</li>
    <li>caprine-bin</li>
    <li>cataclysm-dda</li>
    <li>cataclysm-dda-git</li>
    <li>cdogs-sdl</li>
    <li>chromium-bsu</li>
    <li>chrysalis</li>
    <li>cider</li>
    <li>cloud-init</li>
    <li>cloud-utils</li>
    <li>colima</li>
    <li>colobot</li>
    <li>commandergenius</li>
    <li>cozy-drive</li>
    <li>crawlTiles</li>
    <li>crypto-org-wallet</li>
    <li>devdocs-desktop</li>
    <li>devilutionx</li>
    <li>diffoscope</li>
    <li>dxx-rebirth</li>
    <li>electron-mail</li>
    <li>electronplayer</li>
    <li>endgame-singularity</li>
    <li>enigma</li>
    <li>fheroes2</li>
    <li>firefly-desktop</li>
    <li>flare</li>
    <li>flexoptix-app</li>
    <li>fluent-reader</li>
    <li>framesh</li>
    <li>freeciv</li>
    <li>freeciv_qt</li>
    <li>freeciv_sdl2</li>
    <li>freenukum</li>
    <li>frescobaldi</li>
    <li>frogatto</li>
    <li>fspy</li>
    <li>gnome.gnome-boxes</li>
    <li>gource</li>
    <li>guestfs-tools</li>
    <li>guile-sdl2</li>
    <li>hedgewars</li>
    <li>hikounomizu</li>
    <li>infra-arcana</li>
    <li>instead</li>
    <li>instead-launcher</li>
    <li>irccloud</li>
    <li>joplin-desktop</li>
    <li>keeperrl</li>
    <li>keet</li>
    <li>klystrack</li>
    <li>koboredux-free</li>
    <li>kodiPackages.steam-launcher</li>
    <li>lbreakouthd</li>
    <li>lbry</li>
    <li>ldtk</li>
    <li>ledger-live-desktop</li>
    <li>lens</li>
    <li>libguestfs</li>
    <li>libguestfs-with-appliance</li>
    <li>libresprite</li>
    <li>lima</li>
    <li>lima-bin</li>
    <li>lincity_ng</li>
    <li>linthesia</li>
    <li>linuxKernel.packages.linux_4_14.virtualbox</li>
    <li>linuxKernel.packages.linux_4_14_hardened.virtualbox</li>
    <li>linuxKernel.packages.linux_4_19.virtualbox</li>
    <li>linuxKernel.packages.linux_4_19_hardened.virtualbox</li>
    <li>linuxKernel.packages.linux_5_10.virtualbox</li>
    <li>linuxKernel.packages.linux_5_10_hardened.virtualbox</li>
    <li>linuxKernel.packages.linux_5_15.virtualbox</li>
    <li>linuxKernel.packages.linux_hardened.virtualbox (linuxKernel.packages.linux_5_15_hardened.virtualbox)</li>
    <li>linuxKernel.packages.linux_5_4.virtualbox</li>
    <li>linuxKernel.packages.linux_6_1.virtualbox</li>
    <li>linuxKernel.packages.linux_6_1_hardened.virtualbox</li>
    <li>linuxKernel.packages.linux_latest_libre.virtualbox</li>
    <li>linuxKernel.packages.linux_libre.virtualbox</li>
    <li>linuxKernel.packages.linux_lqx.virtualbox</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.virtualbox</li>
    <li>linuxKernel.packages.linux_xanmod.virtualbox</li>
    <li>linuxKernel.packages.linux_xanmod_latest.virtualbox (linuxKernel.packages.linux_xanmod_stable.virtualbox)</li>
    <li>linuxKernel.packages.linux_zen.virtualbox</li>
    <li>liquidsoap</li>
    <li>logseq</li>
    <li>logstalgia</li>
    <li>losslesscut-bin</li>
    <li>lpairs2</li>
    <li>lunar-client</li>
    <li>lutris</li>
    <li>manaplus</li>
    <li>marktext</li>
    <li>mathpix-snipping-tool</li>
    <li>mendeley</li>
    <li>mepo</li>
    <li>methane</li>
    <li>minigalaxy</li>
    <li>mobilecoin-wallet</li>
    <li>molotov</li>
    <li>motrix</li>
    <li>multipass</li>
    <li>museeks</li>
    <li>mycrypto</li>
    <li>neo4j-desktop</li>
    <li>notable</li>
    <li>notion-app-enhanced</li>
    <li>nsz (python310Packages.nsz)</li>
    <li>nuclear</li>
    <li>ocamlPackages.tsdl-image</li>
    <li>octant-desktop</li>
    <li>open-watcom-bin</li>
    <li>open-watcom-bin-unwrapped</li>
    <li>opendune</li>
    <li>openmsx</li>
    <li>openspades</li>
    <li>osu-lazer-bin</li>
    <li>out-of-tree</li>
    <li>p3x-onenote</li>
    <li>passky-desktop</li>
    <li>picoloop</li>
    <li>pingus</li>
    <li>pioneer</li>
    <li>playonlinux</li>
    <li>plexamp</li>
    <li>polypane</li>
    <li>prboom-plus</li>
    <li>protontricks</li>
    <li>pysolfc</li>
    <li>python310Packages.guestfs</li>
    <li>python310Packages.kivy</li>
    <li>python310Packages.pyautogui</li>
    <li>python310Packages.pygame</li>
    <li>python310Packages.pygame-gui</li>
    <li>python310Packages.pygame_sdl2</li>
    <li>python310Packages.pygetwindow</li>
    <li>python310Packages.pyrect</li>
    <li>python310Packages.pysdl2</li>
    <li>python310Packages.pytmx</li>
    <li>python311Packages.guestfs</li>
    <li>python311Packages.kivy</li>
    <li>python311Packages.nsz</li>
    <li>python311Packages.pygame_sdl2</li>
    <li>python311Packages.pysdl2</li>
    <li>qemu</li>
    <li>qemu-utils</li>
    <li>qemu_full</li>
    <li>qemu_kvm</li>
    <li>qtemu</li>
    <li>quickemu</li>
    <li>radicle-upstream</li>
    <li>rambox</li>
    <li>raven-reader</li>
    <li>redeclipse</li>
    <li>remnote</li>
    <li>renpy</li>
    <li>rocksndiamonds</li>
    <li>saleae-logic-2</li>
    <li>sauerbraten</li>
    <li>sdlpop</li>
    <li>session-desktop</li>
    <li>solarus</li>
    <li>solarus-quest-editor</li>
    <li>sonixd</li>
    <li>ssb-patchwork</li>
    <li>standardnotes</li>
    <li>station</li>
    <li>steam</li>
    <li>steam-rom-manager</li>
    <li>steam-run</li>
    <li>steam-tui</li>
    <li>steamcmd</li>
    <li>steamos-devkit</li>
    <li>superTux</li>
    <li>tauon</li>
    <li>the-legend-of-edgar</li>
    <li>timeular</li>
    <li>tome4</li>
    <li>trackballs</li>
    <li>trezor-suite</li>
    <li>trigger</li>
    <li>tusk</li>
    <li>uhk-agent</li>
    <li>uhk-udev-rules</li>
    <li>ultrastardx</li>
    <li>unityhub</li>
    <li>vagrant</li>
    <li>vcmi</li>
    <li>vdrift</li>
    <li>vdrift-bin</li>
    <li>via</li>
    <li>vial</li>
    <li>video2midi</li>
    <li>virtualbox</li>
    <li>virtualboxHardened</li>
    <li>virtualboxHeadless</li>
    <li>virtualboxWithExtpack</li>
    <li>wargus</li>
    <li>wesnoth</li>
    <li>widelands</li>
    <li>wootility</li>
    <li>wpsoffice</li>
    <li>wpsoffice-cn</li>
    <li>xivlauncher</li>
    <li>youtube-music</li>
    <li>ytmdesktop</li>
    <li>zecwallet-lite</li>
    <li>zettlr</li>
    <li>zpool-auto-expand-partitions</li>
    <li>zulip</li>
  </ul>
</details>

**Comment by superherointj** [comment]
Handled all issues (so far). Feel free to review/merge.

**Comment by superherointj** [comment]
Result of `nixpkgs-review pr 215200` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>aqemu</li>
    <li>python310Packages.rl-coach</li>
  </ul>
</details>
<details>
  <summary>1 package blacklisted:</summary>
  <ul>
    <li>appimage-run-tests</li>
  </ul>
</details>
<details>
  <summary>14 packages failed to build:</summary>
  <ul>
    <li>cot (python310Packages.cot)</li>
    <li>d1x-rebirth-full</li>
    <li>d2x-rebirth-full</li>
    <li>koboredux</li>
    <li>ocamlPackages.ocaml-freestanding</li>
    <li>pureref</li>
    <li>python311Packages.cot</li>
    <li>python311Packages.pyautogui</li>
    <li>python311Packages.pygame</li>
    <li>python311Packages.pygame-gui</li>
    <li>python311Packages.pygetwindow</li>
    <li>python311Packages.pyrect</li>
    <li>python311Packages.pytmx</li>
    <li>solo5</li>
  </ul>
</details>
<details>
  <summary>239 packages built:</summary>
  <ul>
    <li>SDL2_image</li>
    <li>SDL2_image_2_0_5</li>
    <li>Sylk</li>
    <li>_20kly</li>
    <li>abbaye-des-morts</li>
    <li>alephone</li>
    <li>alephone-durandal</li>
    <li>alephone-eternal</li>
    <li>alephone-evil</li>
    <li>alephone-infinity</li>
    <li>alephone-marathon</li>
    <li>alephone-pathways-into-darkness</li>
    <li>alephone-pheonix</li>
    <li>alephone-red</li>
    <li>alephone-rubicon-x</li>
    <li>alpine-make-vm-image</li>
    <li>altair</li>
    <li>anbox</li>
    <li>anki-bin</li>
    <li>anytype</li>
    <li>appimage-run</li>
    <li>assaultcube</li>
    <li>beekeeper-studio</li>
    <li>betterdiscord-installer</li>
    <li>blobby</li>
    <li>blobwars</li>
    <li>blockattack</li>
    <li>bloomrpc</li>
    <li>bootstrap-studio</li>
    <li>brutalmaze</li>
    <li>buttercup-desktop</li>
    <li>caprine-bin</li>
    <li>cataclysm-dda</li>
    <li>cataclysm-dda-git</li>
    <li>cdogs-sdl</li>
    <li>chromium-bsu</li>
    <li>chrysalis</li>
    <li>cider</li>
    <li>cloud-init</li>
    <li>cloud-utils</li>
    <li>colima</li>
    <li>colobot</li>
    <li>commandergenius</li>
    <li>cozy-drive</li>
    <li>crawlTiles</li>
    <li>crypto-org-wallet</li>
    <li>devdocs-desktop</li>
    <li>devilutionx</li>
    <li>diffoscope</li>
    <li>dxx-rebirth</li>
    <li>electron-mail</li>
    <li>electronplayer</li>
    <li>endgame-singularity</li>
    <li>enigma</li>
    <li>fheroes2</li>
    <li>firefly-desktop</li>
    <li>flare</li>
    <li>flexoptix-app</li>
    <li>fluent-reader</li>
    <li>framesh</li>
    <li>freeciv</li>
    <li>freeciv_qt</li>
    <li>freeciv_sdl2</li>
    <li>freenukum</li>
    <li>frescobaldi</li>
    <li>frogatto</li>
    <li>fspy</li>
    <li>gnome.gnome-boxes</li>
    <li>gource</li>
    <li>guestfs-tools</li>
    <li>guile-sdl2</li>
    <li>hedgewars</li>
    <li>hikounomizu</li>
    <li>infra-arcana</li>
    <li>instead</li>
    <li>instead-launcher</li>
    <li>irccloud</li>
    <li>joplin-desktop</li>
    <li>keeperrl</li>
    <li>keet</li>
    <li>klystrack</li>
    <li>koboredux-free</li>
    <li>kodiPackages.steam-launcher</li>
    <li>lbreakouthd</li>
    <li>lbry</li>
    <li>ldtk</li>
    <li>ledger-live-desktop</li>
    <li>lens</li>
    <li>libguestfs</li>
    <li>libguestfs-with-appliance</li>
    <li>libresprite</li>
    <li>lima</li>
    <li>lima-bin</li>
    <li>lincity_ng</li>
    <li>linthesia</li>
    <li>linuxKernel.packages.linux_4_14.virtualbox</li>
    <li>linuxKernel.packages.linux_4_14_hardened.virtualbox</li>
    <li>linuxKernel.packages.linux_4_19.virtualbox</li>
    <li>linuxKernel.packages.linux_4_19_hardened.virtualbox</li>
    <li>linuxKernel.packages.linux_5_10.virtualbox</li>
    <li>linuxKernel.packages.linux_5_10_hardened.virtualbox</li>
    <li>linuxKernel.packages.linux_5_15.virtualbox</li>
    <li>linuxKernel.packages.linux_hardened.virtualbox (linuxKernel.packages.linux_5_15_hardened.virtualbox)</li>
    <li>linuxKernel.packages.linux_5_4.virtualbox</li>
    <li>linuxKernel.packages.linux_6_1.virtualbox</li>
    <li>linuxKernel.packages.linux_6_1_hardened.virtualbox</li>
    <li>linuxKernel.packages.linux_latest_libre.virtualbox</li>
    <li>linuxKernel.packages.linux_libre.virtualbox</li>
    <li>linuxKernel.packages.linux_lqx.virtualbox</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.virtualbox</li>
    <li>linuxKernel.packages.linux_xanmod.virtualbox</li>
    <li>linuxKernel.packages.linux_xanmod_latest.virtualbox (linuxKernel.packages.linux_xanmod_stable.virtualbox)</li>
    <li>linuxKernel.packages.linux_zen.virtualbox</li>
    <li>liquidsoap</li>
    <li>logseq</li>
    <li>logstalgia</li>
    <li>losslesscut-bin</li>
    <li>lpairs2</li>
    <li>lunar-client</li>
    <li>lutris</li>
    <li>manaplus</li>
    <li>marktext</li>
    <li>mathpix-snipping-tool</li>
    <li>mendeley</li>
    <li>mepo</li>
    <li>methane</li>
    <li>minigalaxy</li>
    <li>mobilecoin-wallet</li>
    <li>molotov</li>
    <li>motrix</li>
    <li>multipass</li>
    <li>museeks</li>
    <li>mycrypto</li>
    <li>neo4j-desktop</li>
    <li>notable</li>
    <li>notion-app-enhanced</li>
    <li>nsz (python310Packages.nsz)</li>
    <li>nuclear</li>
    <li>ocamlPackages.tsdl-image</li>
    <li>octant-desktop</li>
    <li>open-watcom-bin</li>
    <li>open-watcom-bin-unwrapped</li>
    <li>opendune</li>
    <li>openmsx</li>
    <li>openspades</li>
    <li>osu-lazer-bin</li>
    <li>out-of-tree</li>
    <li>p3x-onenote</li>
    <li>passky-desktop</li>
    <li>picoloop</li>
    <li>pingus</li>
    <li>pioneer</li>
    <li>playonlinux</li>
    <li>plexamp</li>
    <li>polypane</li>
    <li>prboom-plus</li>
    <li>protontricks</li>
    <li>pysolfc</li>
    <li>python310Packages.guestfs</li>
    <li>python310Packages.kivy</li>
    <li>python310Packages.pyautogui</li>
    <li>python310Packages.pygame</li>
    <li>python310Packages.pygame-gui</li>
    <li>python310Packages.pygame_sdl2</li>
    <li>python310Packages.pygetwindow</li>
    <li>python310Packages.pyrect</li>
    <li>python310Packages.pysdl2</li>
    <li>python310Packages.pytmx</li>
    <li>python311Packages.guestfs</li>
    <li>python311Packages.kivy</li>
    <li>python311Packages.nsz</li>
    <li>python311Packages.pygame_sdl2</li>
    <li>python311Packages.pysdl2</li>
    <li>qemu</li>
    <li>qemu-utils</li>
    <li>qemu_full</li>
    <li>qemu_kvm</li>
    <li>qtemu</li>
    <li>quickemu</li>
    <li>radicle-upstream</li>
    <li>rambox</li>
    <li>raven-reader</li>
    <li>redeclipse</li>
    <li>remnote</li>
    <li>renpy</li>
    <li>rocksndiamonds</li>
    <li>saleae-logic-2</li>
    <li>sauerbraten</li>
    <li>sdlpop</li>
    <li>session-desktop</li>
    <li>solarus</li>
    <li>solarus-quest-editor</li>
    <li>sonixd</li>
    <li>ssb-patchwork</li>
    <li>standardnotes</li>
    <li>station</li>
    <li>steam</li>
    <li>steam-rom-manager</li>
    <li>steam-run</li>
    <li>steam-tui</li>
    <li>steamcmd</li>
    <li>steamos-devkit</li>
    <li>superTux</li>
    <li>tauon</li>
    <li>the-legend-of-edgar</li>
    <li>timeular</li>
    <li>tome4</li>
    <li>trackballs</li>
    <li>trezor-suite</li>
    <li>trigger</li>
    <li>tusk</li>
    <li>uhk-agent</li>
    <li>uhk-udev-rules</li>
    <li>ultrastardx</li>
    <li>unityhub</li>
    <li>vagrant</li>
    <li>vcmi</li>
    <li>vdrift</li>
    <li>vdrift-bin</li>
    <li>via</li>
    <li>vial</li>
    <li>video2midi</li>
    <li>virtualbox</li>
    <li>virtualboxHardened</li>
    <li>virtualboxHeadless</li>
    <li>virtualboxWithExtpack</li>
    <li>wargus</li>
    <li>wesnoth</li>
    <li>widelands</li>
    <li>wootility</li>
    <li>wpsoffice</li>
    <li>wpsoffice-cn</li>
    <li>xivlauncher</li>
    <li>youtube-music</li>
    <li>ytmdesktop</li>
    <li>zecwallet-lite</li>
    <li>zettlr</li>
    <li>zpool-auto-expand-partitions</li>
    <li>zulip</li>
  </ul>
</details>

File: pkgs/development/python-modules/pygame/default.nix [hunk 1]
```diff
@@ -1,5 +1,5 @@
 { stdenv, lib, substituteAll, fetchFromGitHub, buildPythonPackage, python, pkg-config, libX11
-, SDL2, SDL2_image, SDL2_mixer, SDL2_ttf, libpng, libjpeg, portmidi, freetype, fontconfig
+, SDL2, SDL2_image_2_0_5, SDL2_mixer, SDL2_ttf, libpng, libjpeg, portmidi, freetype, fontconfig
```
Comment (PedroHLC) [inline pkgs/development/python-modules/pygame/default.nix:2]: I think it's a common pattern to enforce it in all-packages instead.

File: pkgs/games/toppler/default.nix [hunk 2]
```diff
@@ -10,7 +10,7 @@
 
 , SDL2
 , SDL2_mixer
-, SDL2_image
+, SDL2_image_2_0_5
```
Comment (PedroHLC) [inline pkgs/games/toppler/default.nix:13]: same as above.

Comment (superherointj) [reply pkgs/games/toppler/default.nix]: Thanks. Fixed.

**Review by PedroHLC — APPROVED** [review]
diff LGTM

**Pull request merged**